### PR TITLE
bpo-38119: Fix shmem resource tracking

### PR DESCRIPTION
**Pull request by applio**
Eliminates the use of the resource tracker on shared memory segments but preserves its use on semaphores, etc. where it is appropriately applied.

Adds a test that fails with the current code (an independent Python process started via subprocess accesses shared memory created by a different Python process, then exits normally, expecting the original creator process to continue to be able to access that shared memory) but passes with this patch to defend against future regressions.

https://bugs.python.org/issue38119

### CODE REVIEW

**Comment by vinay0410** [comment]
[Updated PR](https://github.com/python/cpython/pull/21516) fixing merge conflicts.

**Review by col-in-coding — APPROVED** [review]

**Comment by vinay0410** [comment]
@lihaoxiang1989 , this PR has merge conflicts, I have created a PR fixing those, containing the same changes. Can you approve that instead.
[Updated PR](https://github.com/python/cpython/pull/21516) fixing merge conflicts.

**Comment by lthakur007** [comment]
Hello Team,
As am facing the same issue for which this PR has a solution.

May I know, once the PR is merged, how can I get the fix on Ubuntu 18.04?

**Comment by vinay0410** [comment]
@lthakur007 , these changes are now being tracked by https://github.com/python/cpython/pull/21516, once those changes are merged you can either wait for an alpha or beta tag release, or you can directly build python from source (master).

**Comment by lthakur007** [comment]
> @lthakur007 , these changes are now being tracked by #21516, once those changes are merged you can either wait for an alpha or beta tag release, or you can directly build python from source (master).

Thank you for the response.

**Review by zhuowater — COMMENTED** [review]
good idea,it works,nice job!

**Comment by atarashansky** [comment]
Astonishing that this is still not merged... I've been manually unregistering shared memory from child processes for the last year. What's the point of shared memory if it doesn't persist between processes by default...

**Comment by buzmeg** [comment]
@jcarlitooliveira Does #82300 help you in any way?

**Comment by gpshead** [comment]
there's an updated PR mentioned above: https://github.com/python/cpython/pull/21516 - it has a lot more discussion on it so i'm closing this earlier version of the proposed change.

**Pull request closed without merging**
